docs/source/surround.rst

```diff
@@ -1,28 +1,111 @@
-API Documentation
+.. _surround-api:
+
+API
 *****************
+Here you can find documentation on all classes and their methods in Surround.
 
-.. automodule:: surround
+.. _assembler:   
 
-Surround
+Assembler
 ========
 
-.. autoclass:: Surround
-   :members:
+.. autoclass:: surround.assembler.Assembler
+    :members:
+
+.. automodule:: surround
+
+    Config
+    ^^^^^^
+    
+    .. autoclass:: surround.config.Config       
+        :members:          
+ 
+        .. automodule:: surround  
+ 
+    SurroundData                              
+    ^^^^^^^^^^^^   
+                
+    .. autoclass:: surround.SurroundData        
+        :members:                   
+
+    .. automodule:: surround.stage
+
+    Stage
+    ^^^^^
+    
+    .. autoclass:: surround.stage.Stage
+        :members:      
+
+    Validator
+    ^^^^^^^^^
+
+    .. autoclass:: surround.stage.Validator
+        :members:
+
+    Filter
+    ^^^^^^
+
+    .. autoclass:: surround.stage.Filter
+        :members:
 
-SurroundData
-============
+    Estimator
+    ^^^^^^^^^
 
-.. autoclass:: SurroundData
-   :members:
+    .. autoclass:: surround.stage.Estimator
+        :members:
 
-Stage
-=====
+    Visualiser
+    ^^^^^^^^^^
 
-.. autoclass:: Stage
-   :members:
+    .. autoclass:: surround.visualiser.Visualiser
+        :members: 
 
-Config
+Runner
+=======
+.. autoclass:: surround.runners.Runner  
+    :members:
+
+.. _linter:
+
+Linter
```

Comment:
The `Linter` class and functions doesn't need to be documented yet as there isn't much there. I'd rather document it when we want people to look at it and use it.